rpdrtools/io/io.py

```diff
@@ -0,0 +1,201 @@
+import csv
+import logging
+from pathlib import Path
+from typing import Generator, List, Union
+
+import pandas as pd
+from rich.progress import Progress
+from ..constants import (
+    RPDR_NEWLINE_CHAR,
+    RPDR_FILE_DELIMITER,
+    RPDR_REPORT_TEXT_FIELD,
+    RPDR_REPORT_END_TOKEN,
+)
+from .types import OnBrokenRecordsType
+
+
+def _merge_rows(
+    row1: List[str], row2: List[str], newline_char: str = RPDR_NEWLINE_CHAR
+) -> List[str]:
+    """For combining two rows of a "malformed" record when stepping through an
+    RPDR *.txt file.
+
+    This situation mainly occurs when there is an unexpected line break in the
+    middle of a string/text field (e.g. Report_Text).
+
+    The lists are merged by concatenating the two with their "innermost"
+    elements (strings) concatenated, separated by a newline.
+
+    Example:
+
+        Expected number of fields: 4
+
+        ```python
+        row1 = ["A", "B", "C"]
+        row2 = ["D", "E"]
+
+        result = ["A", "B", "C\r\nD", "E"]
+        ```
+    """
+
+    if len(row1) == 0:
+        raise ValueError("First row should not be empty.")
+
+    if len(row2) == 0:
+        row2 = [""]
+
+    return [*row1[:-1], newline_char.join([row1[-1], row2[0]]), *row2[1:]]
+
+
+def _get_bytes(row: List[str], newline_char: str = RPDR_NEWLINE_CHAR) -> int:
+    """Return the number of bytes of a given row as read by csv.reader."""
+
+    return len(("|".join(row) + newline_char).encode("utf-8"))
+
+
+def reader(
+    path: Union[Path, str],
+    on_broken_records: OnBrokenRecordsType = "repair",
+    include_header: bool = False,
+    newline_char: str = RPDR_NEWLINE_CHAR,
+) -> Generator[List[str], None, None]:
+    """Returns a generator that yields the records of a RPDR *.txt file
+    as a list of strings.
+
+    Currently, RPDR files cannot be correctly read by regular CSV tools (e.g.
+    csv.reader, pd.read_csv).
+
+    Due to some formatting quirks with line breaks/quoting, records may be
+    split up over several rows within the CSV.
+
+    If on_broken_records is True, this function attempt to piece together the
```

Comment:
```suggestion
    If on_broken_records is "repair", this function attempts to piece together the
```

I might also add a list of what OnBrokenRecordsType can be in the docstring so the user doesn't have to look it up themselves.